src/js/dragging.js

```diff
@@ -0,0 +1,82 @@
+export function drag() {
+    let leftPane = document.getElementById('left_pane')
+    let rightPane = document.getElementById('right_pane')
+    let separator = document.getElementById('separator')
+
+    separator.style.cursor = 'move'
+
+    let leftLimit = 10
+    let rightLimit = 90
+
+    function sdrag(onDrag, onStop, direction) {
+        let startX = 0
+        let startY = 0
+        let el = separator
+        let dragging = false
+
+        function dragstart() {
+            dragging = true
+        }
+    
+        function dragmove(e) {
+            if (dragging) {
+                let fix = {}
+                    onDrag && onDrag(el, e.pageX, startX, e.pageY, startY, fix)
+                    if (direction !== 'vertical') {
+                        let pageX = ('pageX' in fix) ? fix.pageX : e.pageX
+                        if ('startX' in fix) {
+                            startX = fix.startX
+                        }
+                        if (!('skipX' in fix)) {
+                            el.style.left = (pageX - startX) + 'px'
+                        }
+                    }
+                    if (direction !== 'horizontal') {
+                        let pageY = ('pageY' in fix) ? fix.pageY : e.pageY
+                        if ('startY' in fix) {
+                            startY = fix.startY
+                        }
+                        if (!('skipY' in fix)) {
+                            el.style.top = (pageY - startY) + 'px'
+                        }
+                    }
+            }
+        }
+    
+        function dragend() {
+            dragging = false
```

Comment:
maybe instead of relying on flag, we could add/remove listener for `mousemove` event?